crates/biome_graphql_analyze/src/lint/nursery/use_unique_variable_names.rs

```diff
@@ -0,0 +1,83 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_graphql_syntax::GraphqlVariableDefinitions;
+use biome_rowan::AstNode;
+use biome_rule_options::use_unique_variable_names::UseUniqueVariableNamesOptions;
+
+declare_lint_rule! {
+    /// Require all variable definitions to be unique.
+    ///
+    /// A GraphQL operation is only valid if all its variables are uniquely named.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```graphql,expect_diagnostic
+    /// query C($x: Int, $x: Int) {
+    ///   __typename
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```graphql
+    /// query C($x: Int, $y: Int) {
+    ///   __typename
+    /// }
+    /// ```
+    ///
+    pub UseUniqueVariableNames {
+        version: "next",
+        name: "useUniqueVariableNames",
+        language: "graphql",
+        recommended: false,
+        sources: &[RuleSource::EslintGraphql("unique-variable-names").same()],
+    }
+}
+
+impl Rule for UseUniqueVariableNames {
+    type Query = Ast<GraphqlVariableDefinitions>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseUniqueVariableNamesOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+        let mut found: Vec<String> = Vec::new();
```

Comment:
1. Maybe it would make sense for this to be a `HashSet` or a `FxHashSet`?
2. You should store `TokenText` instead of `String`.